src/metrics/include/metrics/operation.hpp

```diff
@@ -387,6 +392,9 @@ class OperationT final {
     explicit OperationT(internals::OperationImpl<ClockSource>& op) : _op{std::addressof(op)} {}
 
     OperationContextT<ClockSource> start() {
+        BOOST_LOG_TRIVIAL(trace) << "Started collecting metric `" << this->_op->getMetricName()
+                                 << "` for operation `" << this->_op->getOpName() << "` of actor `"
+                                 << this->_op->getActorName() << "`";
```

Comment:
This looks like a lot of overhead for each metric invocation. Does BOOST_LOG_TRIVIAL effectively nop `operator<<` if TRACE isn't enabled? We'd still pay the "cost" of invoking `this->_op->getMetricName()` and the like, but I think we can assume those are trivial (please ensure they're `const` and return refs).

Reply:
Yes it is nop if log level is set higher than TRACE. According to [Boost.Log documentation](https://www.boost.org/doc/libs/1_82_0/libs/log/doc/html/log/tutorial/trivial_filtering.html), it "is only executed if the record passed filtering". Genny sets up filtering [here](https://github.com/mongodb/genny/blob/cfbd1c56e92498e938ada9d1a3aca0a618708d4c/src/driver/src/DefaultDriver.cpp#L159C1-L159C1), and [the default filter level is INFO](https://github.com/mongodb/genny/blob/cfbd1c56e92498e938ada9d1a3aca0a618708d4c/src/driver/src/DefaultDriver.cpp#L376-L378), which is higher than TRACE.

Reply:
Huh: "Remember that the streaming expression is only executed if the record passed filtering." well TIL